addons/payment/models/payment_transaction.py

```diff
@@ -605,6 +605,12 @@ def _create_refund_transaction(self, amount_to_refund=None, **custom_create_valu
             **custom_create_values,
         })
 
+        # Add tokens to refund whenever it exists.
+        if self.token_id and not tx.token_id:
+            tx.token_id = self.token_id
```

Comment:
Small tip: this should be sufficient, since with the actual code, we just created the tx, we know there isn't any token_id set. =]
```suggestion
        tx.token_id = self.token_id
```

And why not pass the value `self.token_id` directly in the create method? :)

Reply:
Thank you, indeed it is sufficient, and better :)

Reply:
Your solution works :)